lib/btrfs_test.pm

```diff
@@ -40,7 +41,34 @@ way to get DBus-less environment is to enter rescue.target via systemctl.
 =cut
 sub snapper_nodbus_setup {
     my ($self) = @_;
-    if (script_run('! systemctl is-active dbus')) {
+    # serial console prints text style as ascii chars in PS1 by default.
+    my $escseq = '[\]\[\(\)\e\d\w\s]*';
+
+    my $rescue_service_config_path = "/usr/lib/systemd/system/rescue.service";
+
+    return unless script_run('! systemctl is-active dbus');
+    if (serial_terminal::select_virtio_console()) {
+
+        # Replace StandardInput in rescue.service file and add TTYPath.
+        script_run "cp $rescue_service_config_path $rescue_service_config_path.sav";
+        script_run "sed -i 's@^StandardInput=.*\$\@StandardInput=tty\\nTTYPath='\$(tty)'\@g' '$rescue_service_config_path'";
+
+        script_run "systemctl daemon-reload";
+        type_string "systemctl rescue";
+        wait_serial "systemctl rescue";
+        type_string "\n";
+        wait_serial ".*or press Control-D to continue.*:";
+        type_password;
+        type_string "\n";
+
+        wait_serial "$escseq:~ #$escseq";
+        type_string 'PS1="# "; export TERM=dumb; stty cols 2048';
+        wait_serial 'PS1="# "; export TERM=dumb; stty cols 2048';
+        type_string "\n";
+        wait_serial "# ";
```

Comment:
Same as @rwx788 I don't understand why we would need to repeat each "type_string" and "wait_serial" call twice each time. Can you not use simply `assert_script_run`?

Reply:
In this case I had indeed issues that this wasn't working reliably. But I will have another try what I can do to get rid of the overhead.